### PR TITLE
Add typings for discontinuous-range

### PR DESCRIPTION
**Pull request by OiCMudkips**
- [X] Prefer to make your PR against the `master` branch.
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [X] Run `npm run lint -- package-name` if a `tslint.json` is present.

If adding a new definition:
- [X] The package does not provide its own types, and you can not add them.
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Run `tsc` without errors.
- [X] Include the required [files](https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package) and header. Base these on the README, *not* on an existing project.

If changing an existing definition:
- [ ] ~Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>~
- [ ] ~Increase the version number in the header if appropriate.~

-----

Adds typings for the npm package [discontinuous-range](https://www.npmjs.com/package/discontinuous-range).

### CODE REVIEW

File: discontinuous-range/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+     * Removes the numbers from rangeStart to rangeEnd to the current range.
+     * If rangeEnd is not defined, only rangeStart is removed from the current range.
+     * @param rangeStart The first number to remove from the range.
+     * @param rangeEnd   The last number to remove from the range. Defaults to rangeStart.
+     * @returns The range on which subtract was called. It does not contain the numbers from rangeStart to rangeEnd.
+     */
+    subtract(rangeStart: number, rangeEnd?: number): DiscontinuousRange;
+
+    /**
+     * Removes the current range from the current range.
```
Comment (andy-ms) [inline discontinuous-range/index.d.ts]: "Removes `rangeToRemove` from the current range."

File: discontinuous-range/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
+// Type definitions for discontinuous-range 1.0
+// Project: https://www.npmjs.com/package/discontinuous-range
+// Definitions by: Victor Zhou <https://github.com/OiCMudkips>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/**
+ * Represents a discontinuous range of numbers.
+ */
+export declare class DiscontinuousRange {
```
Comment (andy-ms) [inline discontinuous-range/index.d.ts]: The library uses `module.exports = DiscontinuousRange;`, which would correspond to `export = DiscontinuousRange; declare class DiscontinuousRange { ... }`. Please verify that these definitions work for you at runtime.

Comment (OiCMudkips) [inline discontinuous-range/index.d.ts]: These definitions worked for me at runtime, but I was only using these typings to import into another typings file. I switched it up to your style, changed my import in my own typings file to `import DiscontinuousRange = require("discontinuous-range");` and it still worked.

File: discontinuous-range/index.d.ts (hunk 1 above)
Comment (OiCMudkips) [inline discontinuous-range/index.d.ts]: Fixed.

**Pull request merged**